java/src/main/java/ai/rapids/cudf/AssertEmptyNulls.java

```diff
@@ -0,0 +1,36 @@
+/*
+ *
+ *  Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package ai.rapids.cudf;
+
+/**
+ *  This class is a Helper class to assert there are no non-empty nulls in a ColumnView
+ *
+ *  The reason for the existence of this class is so that we can turn the asserts on/off when needed
+ *  by passing "-da:ai.rapids.cudf.AssertEmptyNulls". We need that behavior because we have tests
+ *  that explicitly test with ColumnViews that contain non-empty nulls but more importantly, there
```

Comment:
This is exactly why we want the assert in place for tests. CUDF has a goal to never produce non-empty nulls if the input did not contain them.  If we or anyone else is interacting with an external system, we need to remove the non-empty nulls or risk data corruption.  This is here as a early warning to tell us that we have a problem that needs to be fixed. We should fix all of the issues to make this pass before we check it in.

Reply:
@jlowe let's discuss on this thread. 

I get that CUDF has a goal to never produce or deal with non-empty nulls which is why we added this assert. We also have unit-tests in cudf java that introduce empty-nulls for tests. Now we either skip those tests or delete them, or we disable the asserts so we can test the CUDF functionality like `purgeNonEmptyNullsList()`. I chose to do the last option because I think that funcionality should be tested. What am I doing wrong here?

Reply:
I think alternatively we can have these unit-tests in a separate execution tag and disable the assertions only for the 4 tests. 

That sounds better I think

Reply:
That's why I was asking what the goal of the PR is.

> we disable the asserts so we can test the CUDF functionality

By disabling the asserts during all of the tests, we're effectively disabling the asserts all of the time because assertions are off by default.  In practice they're only on during unit tests.  That makes one wonder what purpose the new assertions have.

> I think alternatively we can have these unit-tests in a separate execution tag and disable the assertions only for the 4 tests.

That's one possibility, and that approach at least doesn't avoid ever exercising the new assertions.